app/src/main/java/com/alphawallet/app/ui/widget/holder/WalletSummaryHolder.java

```diff
@@ -211,15 +215,19 @@ private Wallet fetchWallet(Wallet w)
         RealmWalletData realmWallet = realm.where(RealmWalletData.class)
                 .equalTo("address", w.address)
                 .findFirst();
-
         if (realmWallet != null)
         {
             w.balance = realmWallet.getBalance();
             w.ENSname = realmWallet.getENSName();
             w.name = realmWallet.getName();
             w.ENSAvatar = realmWallet.getENSAvatar();
-        }
 
+            if (w.tokens != null)
+            {
+                TestNetHorizontalListAdapter testNetHorizontalListAdapter = new TestNetHorizontalListAdapter(w.tokens);
+                recyclerView.setAdapter(testNetHorizontalListAdapter);
+            }
+        }
         return w;
```

Comment:
this code should be moved into the widget, so the widget call should look like this:

```TokenWithBalanceView tokenBalanceView (maybe  rename to TokensBalanceView)```
...

```tokenBalanceView.populateTokens(tokens);```

where ```tokens``` is the Token[] array.